File: frontend/src/types/vat-rate.ts
export interface VatRate {
  id: string;
  code: string;
  name: string;
  rate: number;
  vat_type: string;
  is_active: boolean;
  valid_from: string | null;
  valid_to: string | null;
}

export interface CreateVatRate {
  code: string;
  name: string;
  rate: number;
  vat_type: string;
  valid_from?: string;
  valid_to?: string;
}

export interface UpdateVatRate {
  code: string;
  name: string;
  rate: number;
  vat_type: string;
  is_active: boolean;
  valid_from?: string;
  valid_to?: string;
}

File: backend/crates/konto-db/src/entities/vat_rate.rs
use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Serialize, Deserialize)]
#[sea_orm(table_name = "vat_rates")]
pub struct Model {
    #[sea_orm(primary_key, auto_increment = false)]
    pub id: String,
    #[sea_orm(unique)]
    pub code: String,
    pub name: String,
    #[sea_orm(column_type = "Decimal(Some((5, 2)))")]
    pub rate: rust_decimal::Decimal,
    pub vat_type: String,
    pub vat_category: String,
    pub is_active: bool,
    pub valid_from: Option<chrono::NaiveDate>,
    pub valid_to: Option<chrono::NaiveDate>,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}
